### PR TITLE
Update GraphQL Modules w/ enhancements

### PR DESCRIPTION
**Pull request by ardatan**
- Update GraphQL Modules and fix typing issues https://github.com/accounts-js/accounts/pull/568
- `@graphql-modules/epoxy` is deprecated, so changed it to `graphql-toolkit`
- Use `Authorization` instead of `accounts-access-token` as default https://github.com/accounts-js/accounts/issues/564#issuecomment-457862215
- Use factory function for `apollo-link`  https://github.com/accounts-js/accounts/pull/520
- Able to exclude user in context, you may not need `context` if you use scoped providers in GraphQL Modules.
- Able to pass extra fields for TypeORM package's createUser
- Update docs
- Fix `func is not a function` error in `@auth` directive.
- Implement `userFieldsFragment` option of graphql-client
- Use tslib instead of duplicating polyfills

### CODE REVIEW

**Comment by pradel** [comment]
@ardatan if we set `skipLibCheck` to true it means that we have some errors in the types. All our users will have to set `skipLibCheck` to true also which is pretty bad unfortunately :(

**Comment by ardatan** [comment]
@pradel It looks like it is a bug related to TypeScript, because we don't get any errors in GraphQL-Modules. You can see CI passes in our repo.

**Comment by pradel** [comment]
@ardatan I can take a look to fix the issue if you want :)

**Comment by pradel** [comment]
@ardatan Cool, will take a look a bit later today!

**Comment by pradel** [comment]
@ardatan after taking a look I found out that you don't use the `strict` option in your tsconfig file. If I activate the option on the di package on the graphql-modules repository it throw the same error as here.

Ref for the strict option:
```
Enable all strict type checking options. 
Enabling --strict enables --noImplicitAny, --noImplicitThis, --alwaysStrict, --strictBindCallApply, --strictNullChecks, --strictFunctionTypes and --strictPropertyInitialization.
```

It also means that all the users that have the `strict` option enabled will not be able to use your package without getting these errors (unless they set `skipLibCheck` to true).

**Comment by ardatan** [comment]
@pradel Yes, you’re right :) I missed that. I had to put ‘any’ in the signature of that function, which looks ugly. Nevermind, I published a new version that respect user’s choice about ‘strict’. I hope CI will pass

File: packages/rest-express/src/user-loader.ts [hunk 1]
```diff
@@ -7,8 +7,11 @@ export const userLoader = (accountsServer: AccountsServer) => async (
   res: express.Response,
   next: any
 ) => {
-  const accessToken =
-    get(req.headers, 'accounts-access-token') || get(req.body, 'accessToken', undefined);
+  let accessToken =
+    get(req.headers, 'Authorization') ||
+    get(req.headers, 'authorization') ||
+    get(req.body, 'accessToken', undefined);
```
[the remark this answers is not in the document]
Comment (ardatan) [reply packages/rest-express/src/user-loader.ts]: I can remove it if you want?

**Comment by ardatan** [comment]
By the way, we will use `accounts-js` in our WhatsApp clone. So, I really want to use `AccountsLink` and other stuff in this PR. :)

File: packages/e2e/package.json [hunk 2]
```diff
@@ -42,18 +42,22 @@
...
     "express": "4.16.4",
-    "graphql": "14.0.2",
+    "graphql": "14.1.1",
     "lodash": "4.17.11",
     "mongoose": "5.4.7",
     "node-fetch": "2.3.0",
     "typeorm": "0.2.11"
+  },
+  "resolutions": { 
+    "core-js": "3.0.0-beta.11"
```
Comment (pradel) [inline packages/e2e/package.json]: Unfortunately resolution does not work on subpackages with workspaces so I think you can remove it :)

**Comment by pradel** [comment]
The pr looks good 👍 
Before we merge can you please remove the `skipLibCheck` options?
I will release a new version shortly after it's merged!

**Comment by ardatan** [comment]
@pradel Done!

**Comment by yanlinaung30** [comment]
> * Update GraphQL Modules and fix typing issues #568
> * `@graphql-modules/epoxy` is deprecated, so changed it to `graphql-toolkit`
> * Use `Authorization` instead of `accounts-access-token` as default [#564 (comment)](https://github.com/accounts-js/accounts/issues/564#issuecomment-457862215)
> * Use factory function for `apollo-link`  #520
> * Able to exclude user in context, you may not need `context` if you use scoped providers in GraphQL Modules.
> * Able to pass extra fields for TypeORM package's createUser
> * Update docs
> * Fix `func is not a function` error in `@auth` directive.
> * Implement `userFieldsFragment` option of graphql-client
> * Use tslib instead of duplicating polyfills

@ardatan 

I have facing the `func is not a function` error in `@auth` directive.

How can i solve this?  Here is my package.json
```
    "@accounts/graphql-api": "^0.29.0",
    "@accounts/mongo": "^0.29.0",
    "@accounts/password": "^0.29.0",
    "@accounts/server": "^0.29.0",
    "@graphql-modules/core": "^0.7.17",
    "@graphql-toolkit/schema-merging": "^0.10.7",
    "apollo-server-express": "^2.18.0",
    "mongoose": "^5.10.0"
```

I followed the following tutorial and it works fine until I add @auth to type or field of type in typeDefs

https://www.apollographql.com/blog/email-password-authentication-with-accounts-js-and-apollo-server/

**Pull request merged**